src/R/Components/Test/PackageManager/PackageManagerIntegrationTest.cs

```diff
@@ -0,0 +1,208 @@
+using System.ComponentModel.Composition.Hosting;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Microsoft.R.Components.InteractiveWorkflow;
+using Microsoft.R.Components.Test.Fakes.InteractiveWindow;
+using Microsoft.R.Host.Client;
+using Microsoft.R.Host.Client.Test.Script;
+using Microsoft.R.Support.Settings;
+using Microsoft.UnitTests.Core.XUnit;
+using Xunit;
+
+namespace Microsoft.R.Components.Test.PackageManager {
+    public class PackageManagerIntegrationTest : IDisposable, IAsyncLifetime {
+        private readonly ExportProvider _exportProvider;
+        private readonly TestRInteractiveWorkflowProvider _workflowProvider;
+        private readonly MethodInfo _testMethod;
+        private readonly TestFilesFixture _testFiles;
+        private readonly string _repo1Path;
+        private readonly string _libPath;
+        private IRInteractiveWorkflow _workflow;
+
+        public PackageManagerIntegrationTest(RComponentsMefCatalogFixture catalog, TestMethodFixture testMethod, TestFilesFixture testFiles) {
+            _exportProvider = catalog.CreateExportProvider();
+            _workflowProvider = _exportProvider.GetExportedValue<TestRInteractiveWorkflowProvider>();
+            _testMethod = testMethod.MethodInfo;
+            _testFiles = testFiles;
+            _workflowProvider.HostClientApp = new RHostClientTestApp();
+            _repo1Path = _testFiles.GetDestinationPath(Path.Combine("Repos", TestRepositories.Repo1));
+            _libPath = Path.Combine(_testFiles.GetDestinationPath("library"), _testMethod.Name);
+            Directory.CreateDirectory(_libPath);
+        }
+
+        [Test]
+        [Category.PackageManager]
+        public async Task AvailablePackagesCranRepo() {
+            var result = await _workflow.Packages.GetAvailablePackagesAsync();
+
+            result.Should().NotBeEmpty();
+
+            var abcPkg = result.SingleOrDefault(pkg => pkg.Package == "abc");
+            abcPkg.Should().NotBeNull();
+            abcPkg.Version.Length.Should().BeGreaterOrEqualTo(0);
+            abcPkg.Depends.IndexOf("abc.data").Should().BeGreaterOrEqualTo(0);
+            abcPkg.License.IndexOf("GPL").Should().BeGreaterOrEqualTo(0);
+            abcPkg.NeedsCompilation.Should().Be("no");
+        }
+
+        [Test]
+        [Category.PackageManager]
+        public async Task AvailablePackagesLocalRepo() {
+            using (var eval = await _workflow.RSession.BeginEvaluationAsync()) {
+                await SetLocalRepoAsync(eval, _repo1Path);
+            }
+
+            var result = await _workflow.Packages.GetAvailablePackagesAsync();
+            result.Should().HaveCount(3);
+
+            var rtvslib1Pkg = result.SingleOrDefault(pkg => pkg.Package == TestPackages.RtvsLib1.Package);
+            rtvslib1Pkg.Should().NotBeNull();
+            rtvslib1Pkg.Version.Should().Be(TestPackages.RtvsLib1.Version);
```

Comment:
Or maybe even just define `Equals` and friends for `RPackage`, and then you can just use `Be`.